care/facility/api/serializers/patient.py

```diff
@@ -105,8 +113,24 @@ def validate_facility(self, value):
 
     def validate(self, attrs):
         validated = super().validate(attrs)
-        if not self.partial and not validated.get("age") and not validated.get("date_of_birth"):
-            raise serializers.ValidationError({"non_field_errors": [f"Either age or date_of_birth should be passed"]})
+        # flag to check whether validation failed
+        err_flag = False
+        errors = ""
+        date_of_birth = validated.get("date_of_birth")
+        if not self.partial and not validated.get("age") and not date_of_birth:
+            err_flag = True
+            errors += "Either age or date_of_birth should be passed."
+
+        phone_number = validated.get("phone_number")
+
+        # Check for uniqueness of mobile number and date of birth
+        if PatientSearch.objects.filter(phone_number=phone_number, date_of_birth=date_of_birth).exists():
```

Comment:
This is good, but we need a unique constraint at a DB level. Please add it, you will have to update the `.save()` of PatientRegistration as well.

Reply:
@anroopak The `save` method is generally used to update values according to our needs if the model is good enough to save. In case we raise ValidationErrors there, users will get a status code 500.
Do we need still need to implement it this way?